numba/tests/test_withlifting.py

```diff
@@ -945,6 +945,56 @@ def case_objmode_cache(x):
             output = x / 10
         return output
 
+    def test_objmode_reflected_list(self):
+        ret_type = typeof([1, 2, 3, 4, 5])
+        @njit
+        def test2():
+            with objmode(out=ret_type):
+                out = [1, 2, 3, 4, 5]
+            return out
+
+        with self.assertRaises(errors.CompilerError) as raises:
+            test2()
+        self.assertIn(
+            ("Objmode context failed. "
+             "Argument 'out' is using unsupported type. "
```

Comment:
Should the exact unsupported Numba type also be reported to the user?

Reply:
yes, done in https://github.com/numba/numba/pull/6733/commits/c2433b6959c0af1623f508ce131dd74007cf2791